src/content/reference/react-dom/components/form.md

```diff
@@ -0,0 +1,433 @@
+
+#### Caveats {/*caveats*/}
+
+* When a function is passed to `action` or `formAction` the HTTP method will be POST regardless of value of the `method` prop.
+
+---
+
+## Usage {/*usage*/}
+
+### Handle form submission on the client {/*handle-form-submission-on-the-client*/}
+
+Pass a function to the `action` prop of form to run the function when the form is submitted. Unlike a conventional HTML action, the form won't call a server endpoint when a client function is passed to `action` prop of form. [`formData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) will be passed to the function as an arrgument so you can access the data submitted by the form.
+
+<Sandpack>
+
+```js App.js
+export default function Search() {
+  function search(formData) {
+    const query = formData.get("query");
+    alert(`You searched for '${query}'`);
+  }
+  return (
+    <form action={search}>
+      <input name="query" />
+      <button type="submit">Search</button>
+    </form>
+  );
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-6db7f4209-20231021",
+    "react-dom": "18.3.0-canary-6db7f4209-20231021",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js",
+  "devDependencies": {}
+}
+```
+
+</Sandpack>
+
+### Handle form submission with a Server Action {/*handle-form-submission-with-a-server-action*/}
+
+Render a `<form>` with a input and submit button. Pass a server action (a function marked with [`'use server'`](/reference/react/use-server) to the `action` prop of form to run the function when the form is submitted.
+
+This is similar to the way forms work when a URL is passed to the `action` prop. The ability to submit forms without JavaScript enables users to submit forms before JavaScript has had a chance to load. This can be benificial to users who have a slow connection, device, or have JavaScript disabled.
+
+You can use hidden form fields to provide data to the `<form>`'s action. The hidden form field data can be access via the [`formData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) argument pass to the function that is passed to the `action` prop of `<form>`.
+
+```jsx
+import { updateCart } from './lib.js';
+
+function AddToCart({productId}) {
+  async function addToCart(formData) {
+    'use server'
+    const productId = formData.get('productId')
+    await updateCart(productId)
+  }
+  return (
+    <form action={addToCart}>
+        <input type="hidden" name="productId" value={productId} />
+        <button type="submit">Add to Cart</button>
+    </form>
+
+  );
+}
+```
+
+In lieu, of using hidden form fields to provide data to the `<form>`'s action, you can call the <CodeStep step={1}>`bind`</CodeStep> method to supply it with extra arguments. This will bind a new argument (<CodeStep step={2}>`productId`</CodeStep>) to the function in addition to the <CodeStep step={3}>`formData`</CodeStep> that is passed as a argument to the function.
```

Comment:
nit extra ,
```suggestion
In lieu of using hidden form fields to provide data to the `<form>`'s action, you can call the <CodeStep step={1}>`bind`</CodeStep> method to supply it with extra arguments. This will bind a new argument (<CodeStep step={2}>`productId`</CodeStep>) to the function in addition to the <CodeStep step={3}>`formData`</CodeStep> that is passed as a argument to the function.
```